manager/aica_django/converters/Knowledge.py

```diff
@@ -845,31 +875,52 @@ def nmap_scan_to_knowledge(
 
     knowledge_nodes = []
     knowledge_relations = []
+    if "runtime" not in scan_results:
+        print("gotcha!", scan_results)
+
+    assert "runtime" in scan_results
 
     scan_time: Optional[datetime.datetime] = dateparser.parse(
         scan_results["runtime"]["time"]
     )
     assert scan_time is not None
 
     # Not needed and make iteration below messy
-    del scan_results["stats"]
-    del scan_results["runtime"]
+    if "stats" in scan_results:
+        del scan_results["stats"]
+
+    if "runtime" in scan_results:
+        del scan_results["runtime"]
 
     my_hostname = socket.gethostname()
-    my_ipv4 = IPv4Address(socket.gethostbyname(my_hostname)).to_knowledge_node()
-    knowledge_nodes.append(my_ipv4)
+    try:
```

Comment:
Ditto comments above about both exceptions hitting, and the ambiguous except clauses.

Reply:
We still need to define which exceptions we expect in the "except" clauses here.